### PR TITLE
Add action compatibility 

### PR DESCRIPTION
**Pull request by IceTank**
First draft for adding task holding and task compatibility checking. This is probably missing a lot of task that should still be added with identifiers. This includes just the task that used createTask.

### CODE REVIEW

**Comment by IceTank** [comment]
Someone might have to check with a few third party plugins if they still work with these changes. And the list of compatible tasks may not be complete at the moment.

**Comment by rom1504** [comment]
I'd appreciate if people could help review this (maybe @u9g ?)

**Comment by extremeheat** [comment]
I think that some of these should throw an exception, they sound like errors in the user logic, for example not using await. 

Otherwise I think it's OK with some added tests to make sure catching incompatible tasks works and some doc for people to update the list in the future

**Comment by rom1504** [comment]
yeah rereading this why not throw instead of await ?
awaiting is handling the task scheduling for the user, which I think we shouldn't do

File: lib/plugins/tasks.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    if (status === true) return
+    onceWithCleanup(bot.tasks.taskUpdate, 'update', {
+      checkCondition: () => {
+        const { status } = taskCompatibleWithRunningTask(taskIdentifier)
+        return status !== false
+      }
+    })
+  }
+
+  bot.taskCompatibleWithRunningTask = taskCompatibleWithRunningTask
```
Comment (rom1504) [inline lib/plugins/tasks.js:71]: If you want to expose that they should be in the doc at API.md

**Comment by u9g** [comment]
Will this effect usage by the end user? If so, please add a new example to the examples folder of how to use this.

**Comment by IceTank** [comment]
> Will this effect usage by the end user? If so, please add a new example to the examples folder of how to use this.

Depends on how strict you want to make the task conflict configuration. Things like inventory actions should not be done at the same time as it can break both actions. But things like moving while fishing for instance can be more relax. I can add an example to how to deal with actions blocking you from doing certain things. 

This will probably be a difficult change for everyone that is not too familiar with how js or mineflayer works. As you can quickly mess up and have things try and run in parallel making functions suddenly throw errors when they worked fine before. So maybe only activate this when something like a strict option is set? What do you guys think @u9g  @rom1504 ?

**Comment by IceTank** [comment]
I am not sure if the current implementation is working. I will add some tests and examples to test it.

**Comment by rom1504** [comment]
failing and old

**Pull request closed without merging**